src/commands/snackify.ts

```diff
@@ -0,0 +1,85 @@
+import { filesystem, GluegunToolbox } from "gluegun"
+import { packager } from "../tools/packager"
+
+module.exports = {
+  alias: ["s"],
+  description: "Snackify Ignite app",
+  run: async (toolbox: GluegunToolbox) => {
+    const { print, prompt, system, patching } = toolbox
+    const { remove } = filesystem
+    const { info } = print
+
+    // getStatus checks if there are any uncommitted changes inside the branch
+    const getStatus = await system.run(`git status -s`)
+
+    if (getStatus === "") {
+      // test if the branch name has substring snack
+      const branchName = await system.run(`git branch --show-current`)
+      if (!branchName.includes("snack")) {
+        info("Creating a new branch for your snack")
+        system.run(`git checkout -b snack-$(date +'%m%d%y')`)
+      }
+    } else {
+      info(
+        "There are uncommitted changes in your current branch, please commit changes before running this",
+      )
+      return
+    }
```

Comment:
Might want to add a `system.which('git')` check to make sure that we have a git repo at all, since not adding git is a possible options with ignite

Reply:
Probably check for existence of `.git` directory, rather than existence of the `git` command.